modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamIT.java

```diff
@@ -2091,9 +2074,7 @@ public void testWriteLoadAndAvgShardSizeIsStoredInABestEffort() throws Exception
         final var createDataStreamRequest = new CreateDataStreamAction.Request(dataStreamName);
         assertAcked(client().execute(CreateDataStreamAction.INSTANCE, createDataStreamRequest).actionGet());
 
-        for (int i = 0; i < 10; i++) {
-            indexDocs(dataStreamName, randomIntBetween(100, 200));
-        }
+        indexDocsAndEnsureThereIsCapturedWriteLoad(dataStreamName);
```

Comment:
There were a couple of windows failures, I think we don't have enough granularity there to ensure that we capture some write load.

Reply:
I am not sure I follow how that could happen just because of granularity. The test does not validate that there is a > 0 write load.

The assertion that failed seems to be for a null write load. AFAICS, that can only happen if we did not get a stats response from one of the shards. Which sounds strange in this test. Did I miss something obvious?

Reply:
Right 🤦, I think that we allocate all shards into 2 nodes and then we don't get any stats back.